corehq/apps/app_manager/suite_xml.py

```diff
@@ -920,8 +926,63 @@ def details(self):
+                                d = load_xmlobject_from_string(
+                                    detail.custom_xml,
+                                    xmlclass=Detail
+                                )
                                 r.append(d)
+
+                            elif detail.use_case_tiles:
+
+                                template_args = {
+                                    "detail_id": self.id_strings.detail(module, detail_type),
+                                    "title_text_id": self.id_strings.detail_title_locale(
+                                        module, detail_type
+                                    )
+                                }
+
+                                # Get field/case property mappings
+                                for template_field in ["header", "top_left", "sex", "bottom_left", "date"]:
+                                    column = None
+                                    for col in detail.columns:
+                                        if col.case_tile_field == template_field:
+                                            column = col
+                                            break
+
+                                    template_args[template_field] = {
+                                        "prop_name": get_column_xpath_generator(
+                                            self.app, module, detail, column
+                                        ).xpath,
+                                        "locale_id": self.id_strings.detail_column_header_locale(
+                                            module, detail_type, column,
+                                        ),
+                                        # Just using default language for now
+                                        # The right thing to do would be to reference the app_strings.txt I think
+                                        "prefix": escape(
+                                            column.header.get(self.app.default_language, "")
+                                        )
+                                    }
+                                    if column.format == "enum":
+                                        template_args[template_field]["enum_keys"] = {}
+                                        for mapping in column.enum:
+                                            key = mapping.key
+                                            template_args[template_field]["enum_keys"][key] = \
+                                                self.id_strings.detail_column_enum_variable(
+                                                    module, detail_type, column, key
+                                                )
+
+                                # Populate the template
+                                with open(os.path.join(
+                                    os.path.dirname(__file__), "case_tile_templates", "tdh.txt"
```

Comment:
Would be better to read this file once outside the double nested `for` loops. But more generally, is this the best practice for opening a template? This also looks like code for a specific customer, which typically does not belong in the `corehq` package.